src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANG = 100;
+    private static final int MAX_INDEX = 4;
+    private static String COLOR;
+    private final int randomInd = new Random().nextInt(MAX_INDEX);
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (randomInd) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), new Random().nextInt(MAX_RANG));
+            case 1:
+                COLOR = colorSupplier.getRandomColor();
+                return new IsoscelesTrapezoid(COLOR, new Random().nextInt(MAX_RANG),
+                        new Random().nextInt(MAX_RANG), new Random().nextInt(MAX_RANG));
+            case 2:
+                COLOR = colorSupplier.getRandomColor();
```

Comment:
it's not necessary to create the COLOR variable every time, you can just pass colorSupplier.getRandomColor() to constructor

Reply:
i can't pass code quality test https://prnt.sc/I5uMWh38ia6u

Reply:
you can make an enter after the comma in places where it's more than 100 symbols

Reply:
fixed in last commit